### PR TITLE
Fetch latest contributors using FlowPHP itself

### PR DESCRIPTION
**Pull request by stloyd**
<h2>Change Log</h2>
<div id="change-log">
  <h4>Added</h4>
  <ul id="added">
    
  </ul> 
  <h4>Fixed</h4>  
  <ul id="fixed">
    
  </ul>
  <h4>Changed</h4>
  <ul id="changed">
    <li>Fetch latest contributors using FlowPHP itself</li>
  </ul>  
  <h4>Removed</h4>
  <ul id="removed">
    
  </ul>
  <h4>Deprecated</h4>
  <ul id="deprecated">
    
  </ul>  
  <h4>Security</h4> 
  <ul id="security">
    
  </ul>     
</div>
<hr/>

<h2>Description</h2>

Zrzut ekranu 2024-02-2 o 13 09 49

### CODE REVIEW

File: src/web/landing/src/Flow/Website/Controller/DefaultController.php [hunk 1]
```diff
@@ -2,120 +2,60 @@
...
 
-class DefaultController extends AbstractController
+final class DefaultController extends AbstractController
 {
+    private readonly string $path;
+
+    public function __construct(
+        private readonly ContributorsUrlFactory $contributorsUrlFactory
+    ) {
+        $this->path = __DIR__ . '/../../../../var/data/latest_contributors.json';
```
Comment (norberttech) [inline src/web/landing/src/Flow/Website/Controller/DefaultController.php]: I would pass here `ContainerBagInterface $parameters` through constructor and then use `$parameters->get('kernel.cache_dir')` or something similar

File: src/web/landing/src/Flow/Website/Controller/DefaultController.php [hunk 2]
```diff
@@ -2,120 +2,60 @@
...
+    }
+
     #[Route('/', name: 'main')]
-    public function main() : Response
+    public function __invoke() : Response
```
Comment (norberttech) [inline src/web/landing/src/Flow/Website/Controller/DefaultController.php]: I think main was ok, we are also going to add few other pages later like for example something that will read documentation from `/docs` folder and dump it as static page or `installation wizard` that will help determine which packages user needs based on his answer

File: src/web/landing/src/Flow/Website/Controller/DefaultController.php [hunk 3]
```diff
@@ -2,120 +2,62 @@
...
+            ->drop('unpacked', 'data')
+            // Filter out bots & limit the end list
+            ->filter(not(ref('login')->endsWith(lit('[bot]'))))
+            ->filter(not(ref('login')->equals(lit('aeon-automation'))))
+            ->limit(24)
+            // Save with overwrite
+            ->mode(SaveMode::Overwrite)
+            ->write(to_json($this->path))
+            // Execute
+            ->run();
```
Comment (norberttech) [inline src/web/landing/src/Flow/Website/Controller/DefaultController.php:51]: I like how flow is used to generate flow landing page contributors ❤️

Comment (norberttech) [reply src/web/landing/src/Flow/Website/Controller/DefaultController.php:51]: but technically speaking you dont even need to write it anywhere, you can literally load it to memory and dump that memory to twig directly

File: src/web/landing/src/Flow/Website/Controller/DefaultController.php [hunk 4]
```diff
@@ -2,120 +2,56 @@
...
+            ->filter(not(ref('login')->equals(lit('aeon-automation'))))
+            ->limit(24)
+            // Store result in memory
+            ->write(to_memory($memory = new ArrayMemory()))
+            // Execute
+            ->run();
 
         return $this->render('main/index.html.twig', [
-            'contributors' => $contributors,
+            'contributors' => $memory->data,
```
Comment (norberttech) [inline src/web/landing/src/Flow/Website/Controller/DefaultController.php]: I think you just found a bug 🙈 `data` should be private and instead you should use here `$memory->dump()`

Comment (stloyd) [reply src/web/landing/src/Flow/Website/Controller/DefaultController.php]: Fix https://github.com/flow-php/flow/pull/952

**Pull request merged**